### PR TITLE
Feat 1.2/fe dashboard auth

### PR DESCRIPTION
**Pull request by danbergelt**
FE dashboard authentication - pushes users to their appropriate dashboard and restricst access against local storage tampering. Encrypts the role into local storage to obfuscate values.

### CODE REVIEW

File: mission-control-client/src/App.js [hunk 1]
```diff
@@ -1,36 +1,53 @@
...
-import PrivateRoute from "./components/PrivateRoute";
+import {
+  Switch,
+  Route,
+  useLocation,
+  Redirect,
+  useHistory
+} from "react-router-dom";
+import PrivateRoute from "./utils/PrivateRoute";
+import SPrivateRoute from "./utils/S_PrivateRoute";
```
Comment (dionys93) [inline mission-control-client/src/App.js]: It's better to be explicit and call it `StudentPrivateRoute` even if it is a bit longer to write.

File: mission-control-client/src/App.js [hunk 2]
```diff
@@ -1,36 +1,53 @@
...
+import APrivateRoute from "./utils/A_PrivateRoute";
```
Comment (dionys93) [inline mission-control-client/src/App.js]: AdminPrivateRoute

File: mission-control-client/src/utils/S_PrivateRoute.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import React from "react";
```
Comment (dionys93) [inline mission-control-client/src/utils/S_PrivateRoute.js]: `StudentPrivateRoute.js`

File: mission-control-client/src/App.js (hunk 2 above)
Comment (danbergelt) [reply mission-control-client/src/App.js]: Done

File: mission-control-client/src/utils/S_PrivateRoute.js (hunk 3 above)
Comment (danbergelt) [reply mission-control-client/src/utils/S_PrivateRoute.js]: Done

**Review by dionys93 — APPROVED** [review]

**Review by BDesselle — APPROVED** [review]

**Review by agordon1123 — APPROVED** [review]

**Pull request merged**